### PR TITLE
Ensure AuthClient is cached

### PR DESCRIPTION
**Pull request by sethvargo**
Closes https://github.com/google-github-actions/upload-cloud-storage/issues/364

/cc @GregoireW @danielbankhead could you please test this?

@verbanicm hold on merge until further testing.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -159,8 +160,27 @@ export interface ClientComputeDestinationOptions {
...
+    await client.storage.authClient.getClient();
+
+    return client;
+  }
+
+  private constructor(opts?: ClientOptions) {
+    const authClient = new GoogleAuth({
+      projectId: opts?.projectID,
+      universeDomain: opts?.universe,
+    });
```
Comment (sethvargo) [inline src/client.ts:180]: @danielbankhead - is this required, or can we use the authClient that `new Storage` creates by default?

**Comment by GregoireW** [comment]
@sethvargo The action is not built here.

**Comment by sethvargo** [comment]
> @sethvargo The action is not built here.

Fixed

**Comment by GregoireW** [comment]
Now I got 2 api when I launch my test with 200 files so this is good.

I notice a real application with ~75 api call for 3500 files, not sure why, but let say it is not related.  it is already a huge improvement.

Thank you!

**Comment by sethvargo** [comment]
@verbanicm ready for review

@danielbankhead - I'd still like your feedback on https://github.com/google-github-actions/upload-cloud-storage/pull/373#discussion_r1720790152.

**Review by verbanicm — APPROVED** [review]

File: src/client.ts (hunk 1 above)
Comment (danielbankhead) [reply src/client.ts:180]: Not required, you can use the one Storage creates

**Pull request merged**